agentstack/generation/tool_generation.py

```diff
@@ -97,7 +96,7 @@ def add_tool(tool_name: str, agents: Optional[list[str]] = []):
             with ToolsInitFile(conf.PATH / TOOLS_INIT_FILENAME) as tools_init:
                 tools_init.add_import_for_tool(tool, agentstack_config.framework)
         except ValidationError as e:
-            print(term_color(f"Error adding tool:\n{e}", 'red'))
+            raise ToolError(f"Error adding tool:\n{e}")
```

Comment:
if i'm not missing something, `ValidationError` will never be surfaced to `tools.py` for the proper logging because we're catching it here and re-raising it as a `ToolError`